### PR TITLE
Implement R3-based geometrically nonlinear, constitutively linear spring element

### PR DESCRIPTION
**Pull request by faisal-bhuiyan**
Implements the task described [here](https://github.com/Exawind/openturbine-project/issues/19#issue-2715956703). The system kernels were implemented here -- leaving the assembly logic to be implemented in a follow up work.

### CODE REVIEW

**Comment by deslaughter** [comment]
I think it looks pretty good. Most of the CI errors are just cleanup.

File: tests/regression_tests/regression/test_springs.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        1.      // Undeformed length
+    )});
+
+    return CreateSprings(springs_input);
+}
+
+TEST(SpringsTest, NodeStateIndices) {
+    const auto springs = SetUpSprings();
+    EXPECT_EQ(springs.node_state_indices(0, 0), 0);
+    EXPECT_EQ(springs.node_state_indices(0, 1), 1);
```
Comment (ddement) [inline tests/regression_tests/regression/test_springs.cpp:35]: These checks access Device data on host and segfault for GPU builds.  Copy them down to host first.

**Comment by ddement** [comment]
Looks good to me other than the GPU segfault mentioned above

File: tests/regression_tests/regression/test_springs.cpp (hunk 1 above)
Comment (faisal-bhuiyan) [reply tests/regression_tests/regression/test_springs.cpp]: Could you confirm if the changes fixed the segfault?

Comment (ddement) [reply tests/regression_tests/regression/test_springs.cpp]: It works now!

**Review by ddement — APPROVED** [review]

**Pull request merged**